extensions/amp-list/0.1/amp-list.js

```diff
@@ -339,9 +340,11 @@ export class AmpList extends AMP.BaseElement {
       current.rejecter();
     };
     if (this.ssrTemplateHelper_.isSupported()) {
-      const json = /** @type {!JsonObject} */ (current.data);
-      this.templates_.findAndRenderTemplate(this.element, json)
-          .then(element => this.container_.appendChild(element))
+      // TODO(alabiaga): This is a misleading type cast. Instead, we should use
+      // a new API on template-impl.js and amp-mustache.js as discussed.
+      const html = /** @type {!JsonObject} */ (current.data);
+      this.templates_.findAndRenderTemplate(this.element, html)
+          .then(element => this.render_([element]))
```

Comment:
@alabiaga I think this was a bug. We need to call `render_` so the old rendered elements are removed, among other things.